### PR TITLE
Specify error message when Certificate URLs and passphrases don't match

### PR DESCRIPTION
**Pull request by Bence1001**
### Checklist
- [x] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- [X] `step.yml` and `README.md` is updated with the changes (if needed)

### Version
Requires a *PATCH* [version update](https://semver.org/)

### Context
* The step can take multiple Certificate URLs and passphrases, separated by pipe symbols (`|`).
* This can be a problem if one of the passphrases contains a pipe symbol.
* Choosing another character (or string) as a separator will not work, as there's no limitation on what character can be in a Certificate passphrase (tested this manually). These passphrases are most likely generated randomly to be secure, which increases the likelihood of random characters that match our chosen separator.
* Currently, we have no better solution for ingesting "infinitely long" list inputs. We have to recognize this as a limitation.
* Therefore, the best we can do here is provide a **more informative error message.**

### Changes
* **Specify error message when Certificate URLs and passphrases don't match**, as one of the reasons could be that passphrases contain pipes.
* Added E2E test for checking if the step fails correctly in this situation.
* I did some cleanup to remove old conventions that we no longer use.

### CODE REVIEW

**Review by shams-ahmed — DISMISSED** [review]

**Comment by shams-ahmed** [comment]
@Bence1001 Can you update the `step.yml` file as well. A sample sentence under `certificate_passphrase` stating the limitations..

File: e2e/bitrise.yml [hunk 1]
```diff
@@ -11,6 +11,29 @@ workflows:
     after_run:
     - _run
 
+  test_certificate_url_and_passphrase_not_equal:
```
[the remark this answers is not in the document]
Comment (Bence1001) [reply e2e/bitrise.yml:14]: You are right, I didn't recognize this at first, but looking back now it is confusing 😅

File: main.go [hunk 2]
```diff
@@ -349,11 +337,16 @@ func main() {
...
 
-		// Do not splitAndTrimSpace passphrases, since passphrase is may empty !!!
+		// Do not splitAndTrimSpace passphrases, since a passphrase might be empty!
 		certificatePassphrases := strings.Split(configs.CertificatePassphrase, "|")
 
 		if len(certificateURLs) != len(certificatePassphrases) {
-			failF("Certificate url count: (%d), not equals to Certificate Passphrase count: (%d)", len(certificateURLs), len(certificatePassphrases))
+			failF(
+				"Certificate URL count: (%d), is not equal to Certificate passphrase count: (%d).\n"+
+					"This could be because one of your passphrases contains a pipe symbol (\"|\") which is forbidden.",
```
[the remark this answers is not in the document]
Comment (Bence1001) [reply main.go:346]: I'll try to come up with something, thanks for the feedback.

[the remark this answers is not in the document]
Comment (Bence1001) [reply main.go]: Let me know if you are OK with the new wording!

**Comment by Bence1001** [comment]
I've regenerated the README + did some cleanup in the `step.yml`.

**Pull request merged**